### ISSUE TITLE
Test the handling of focus when editing entity data

### ISSUE THREAD

**Issue by draganescu** [WordPress/gutenberg#49098]
In #48979 a fix was merged to solve focus loss which occurred when undo or redo was used inside an editor who managed entity editing via `useBlockSync`. We need to add a test that tests for our desired behavior in terms of focus management.

Ideally focus should remain to where it was on actions such as undo and redo. However, #48979 has a temporary - I hope - solution which moves the focus to the parent when the undo changes controlled inner blocks. This should be fixed too, but for now the test should test this focus parent behavior.
